library/src/main/java/pl/netigen/premium/PremiumBuilderNetigenApi.java

```diff
@@ -0,0 +1,137 @@
+package pl.netigen.premium;
+
+import android.view.View;
+
+import java.util.List;
+
+import androidx.annotation.DrawableRes;
+import androidx.annotation.NonNull;
+import androidx.annotation.StringRes;
+import androidx.fragment.app.FragmentManager;
+import androidx.fragment.app.FragmentTransaction;
+
+public class PremiumBuilderNetigenApi {
+
+    int bgImage;
+    int titleImage;
+    int bgPremiumItem;
+    int imageButtonBuy;
+    int premiumOptionsTitle;
+    int textBuyButton;
+    String price;
+    List<PremiumItemNetigenApi> premiumItemNetigenApis;
+    View.OnClickListener clickBuyPremium;
+
+    private PremiumBuilderNetigenApi(int bgImage, int titleImage, int bgPremiumItem, int imageButtonBuy,
+                                     int premiumOptionsTitle, int textBuyButton, String price, List<PremiumItemNetigenApi> premiumItemNetigenApis,
+                                     View.OnClickListener clickBuyPremium) {
+        this.bgImage = bgImage;
+        this.titleImage = titleImage;
+        this.bgPremiumItem = bgPremiumItem;
+        this.imageButtonBuy = imageButtonBuy;
+        this.premiumOptionsTitle = premiumOptionsTitle;
+        this.textBuyButton = textBuyButton;
+        this.price = price;
+        this.premiumItemNetigenApis = premiumItemNetigenApis;
+        this.clickBuyPremium = clickBuyPremium;
+    }
+
+    public void openPremiumFragment(@NonNull FragmentManager fragmentManager,
+                                    int frameId, String profile) {
+        testValues();
+        PremiumFragmentNetigenApi premiumFragmentNetigenApi = PremiumFragmentNetigenApi.newInstance(this);
+        FragmentTransaction transaction = fragmentManager.beginTransaction();
+        transaction.add(frameId, premiumFragmentNetigenApi, profile);
+        transaction.addToBackStack(null);
+        transaction.commit();
+    }
+
+    private void testValues() {
+        if (bgImage == 0) {
```

Comment:
If these values are necessary they should be included in Builder Constructor method.